src/main/java/com/tsmms/skoop/user/UserPermissionRequest.java

```diff
@@ -25,4 +25,7 @@
 
 	@ApiModelProperty("IDs of the users to grant the permission to.")
 	private List<String> authorizedUserIds;
+
+	@ApiModelProperty("Is all users authorized.")
+	private boolean allUsersAuthorized;
```

Comment:
The `UserPermission` was intended to grant specific other users permission to access certain scopes of the owner's personal data. Granting permission to all users is therefore not a `UserPermission` and should not be handled as part of this API. It is rather a global permission similar to the consent to being a coach.

I suggest one of these two approaches to keep global permission separate from the `UserPermission` implementation:

- Add the permission as a `User` property like with the `coach` consent. In this case it could be the boolean flag `publicProfile`.
- Create another node entity named `GlobalPermission` with the field `scope` which is of enum type `GlobalPermissionScope`. Then create relationships between `User` entities and `GlobalPermission` entities with appropriate scopes. We should also provide a new REST API resource `/users/{userId}/global-permissions`. The coach consent could also be refactored to a global permission to keep the `User` node entity clean of any permission-related properties.

I would prefer the second option but it is definitely more work, I know.

Reply:
I would go for the second option either. It is more work but it will make good in future as it is easier to extend to other business-cases. If you don't mind I will change the implementation according to the second approach.

Reply:
Alright, please implement the second approach. 😃 

I would suggest to create individual `GlobalPermission` instances for each user instead of relating users to shared instances. This will allow us to relate global permissions to other user-specific nodes to handle fine-grained access control, e.g. user wants to give all other users permission to view only a specific project she/he has worked on.

By example:

Assuming you and I want to grant all users permission to view our personal profile. In the end we have to create two `GlobalPermission` instances with the scope `READ_USER_PROFILE`. Every user gets related to his own instance of those global permissions.

`(:User {userName: 'dima'}-[:HAS_GRANTED]-(:GlobalPermission {id: 1, scope: 'READ_USER_PROFILE'})`
`(:User {userName: 'georg'}-[:HAS_GRANTED]-(:GlobalPermission {id: 2, scope: 'READ_USER_PROFILE'})`

The REST API could be implemented in a similar fashion like the current user permissions. At the moment we only need one API to read all global permissions and one to replace the whole list of global permissions.

`GET /users/{userId}/global-permissions` returns the list of all global permissions for the given user.
`PUT /users/{userId}/global-permissions` replaces the global permissions for the given user with the ones given in the request body.

Is that OK? Did I miss something?

Reply:
That is OK. Thank you a lot! I think you summarized all we need at the moment.

Reply:
I have implemented the concept of `GlobalPermission`.
The access to user profile is managed by `com.tsmms.skoop.user.UserPermissionScope#READ_USER_PROFILE` permission.
Being a coach is also managed by global permissions with `com.tsmms.skoop.user.UserPermissionScope#SEE_AS_COACH`.